x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -100,108 +109,349 @@ func (s *KeeperTestSuite) SetupGetHostZoneUnbondingMsgs() GetHostZoneUnbondingMs
+
+		if tc.expectPass {
+			s.Require().Equal(int64(amtToUnbond)*int64(len(tc.epochUnbondingRecords)), int64(actualAmtToUnbond), "total amount to unbond should match input amtToUnbond")
+			s.Require().Equal(len(tc.epochUnbondingRecords), len(actualUnbondRecords))
+		} else {
+			// no messages should be sent
+			s.Require().Equal(0, len(actualUnbondRecords), "no messages should be sent")
+			// no value should be unbonded
+			s.Require().Equal(int64(0), int64(actualAmtToUnbond), "no value should be unbonded")
+		}
+	})
+}
+func (s *KeeperTestSuite) TestGetUnbondingAmountAndRecords_WrongChainId() {
+	name := "Wrong chain id"
+	var test = defaultUnbondingTestCase
+	test.chainId = "nonExistentChainId"
+	test.expectPass = false
+	s.VerifyTestCase_TestGetUnbondingAmountAndRecords(name, test)
 
-	// TODO add case that checks the *marshaled* callback args against expectations
+}
 
-	// the number of unbonding messages should be (number of validators) * (records to unbond)
-	s.Require().Equal(len(tc.valNames), len(actualUnbondMsgs), "number of unbonding messages should be number of records to unbond")
+func (s *KeeperTestSuite) TestGetUnbondingAmountAndRecords_Successful() {
+	name := "Successful"
+	var test = defaultUnbondingTestCase
 
-	s.Require().Equal(int64(tc.amtToUnbond)*int64(len(tc.epochUnbondingRecords)), int64(actualAmtToUnbond), "total amount to unbond should match input amtToUnbond")
+	s.VerifyTestCase_TestGetUnbondingAmountAndRecords(name, test)
 
-	totalWgt := sdk.NewDec(int64(tc.totalWgt))
-	actualAmtToUnbondDec := sdk.NewDec(int64(actualAmtToUnbond))
-	actualUnbondMsg1 := actualUnbondMsgs[0].String()
-	actualUnbondMsg2 := actualUnbondMsgs[1].String()
-	val1Fraction := sdk.NewDec(int64(tc.hostZone.Validators[0].Weight)).Quo(totalWgt)
-	val2Fraction := sdk.NewDec(int64(tc.hostZone.Validators[1].Weight)).Quo(totalWgt)
-	val1UnbondAmt := val1Fraction.Mul(actualAmtToUnbondDec).TruncateInt().String()
-	val2UnbondAmt := val2Fraction.Mul(actualAmtToUnbondDec).TruncateInt().String()
+}
+func (s *KeeperTestSuite) TestGetUnbondingAmountAndRecords_NoEpochUnbondingRecords() {
+	name := "No Epoch Unbonding Records"
+	var test = defaultUnbondingTestCase
+	test.expectPass = false
+	test.epochUnbondingRecords = []recordtypes.EpochUnbondingRecord{}
 
-	val1Unbonded := strings.Contains(actualUnbondMsg1, val1UnbondAmt)
-	val2Unbonded := strings.Contains(actualUnbondMsg2, val2UnbondAmt)
+	s.VerifyTestCase_TestGetUnbondingAmountAndRecords(name, test)
 
-	// there's rounding in the logic that distributes stake amongst validators, so one or the other of the balances will be correct, depending on the rounding
-	// at least one will be correct, and the other will be off by 1 by rounding, so we check and OR condition
-	s.Require().True(val1Unbonded || val2Unbonded, "unbonding amt should be the correct amount")
 }
+func (s *KeeperTestSuite) VerifyTestCase_TestDistributeUnbondingAmountToValidators(name string, tc GetHostZoneUnbondingMsgsTestCase) {
```

Comment:
For `DistributeUnbondingAmountToValidators`, we also want to add a test case where a validator does not have enough at stake to handle their share of the unbonding and we traverse [this branch](https://github.com/Stride-Labs/stride/blob/main/x/stakeibc/keeper/unbonding_records.go#L100)

You can test this by setting the delegation amount of one of the validators to 0 (a validator that has non-zero weight) and then checking that the portion that was supposed to be undelegated from that validator was actually split among the others.